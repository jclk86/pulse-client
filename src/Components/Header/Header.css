.header {
  width: 100%;
  background-color: rgb(229, 226, 226);
  margin-bottom: 70px;
}

.container_header_content {
  display: flex;
  max-width: 1400px;
  margin: 0 auto;
}

.navbar_links {
  display: flex;
  width: 50%;
  justify-content: flex-end;
  padding-right: 10px;
  font-size: 16px;
  align-items: center;
  font-size: 20px;
}

.container_logo {
  width: 50%;
  text-align: left;
}

.top_header_logo {
  width: 200px;
}

.navbar_link_signup,
.navbar_link_login {
  text-decoration: none;
  margin-left: 10px;
  color: black;
  font-weight: bold;
}

.navbar_link_signup:hover,
.navbar_link_login:hover,
.navbar_link_account:hover {
  cursor: pointer;
  text-decoration: underline;
}

.navbar_link_account {
  color: black;
  text-decoration: none;
  font-weight: bold;
}

@media only screen and (max-width: 768px) {
  .header {
    margin-bottom: 0;
  }
}

@media only screen and (max-width: 400px) {
  .container_header_logged_in_links,
  .container_header_logged_out_links {
    display: flex;
    flex-direction: column;
    margin-right: 10px;
  }
}
