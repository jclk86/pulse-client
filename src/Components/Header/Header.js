import React, { Component } from "react";
import { NavLink } from "react-router-dom";
import { Logo, Moon } from "../Utils/Utils";
import TokenService from "../../Services/token-service";
import IdleService from "../../Services/idle-service";
import "./Header.css";
import PropTypes from "prop-types";

class Header extends Component {
  static defaultProps = () => {};

  handleLogoutClick = () => {
    TokenService.clearAuthToken();
    TokenService.clearCallbackBeforeExpiry();
    IdleService.unRegisterIdleResets();
  };

  onMoonClick = () => {
    this.props.toggleLights();
  };

  render() {
    return (
      <div
        className={`header ${
          this.props.lightsOff ? "header" : "header_lights_off"
        }`}
      >
        <div className="container_header_content">
          <div className="container_logo">
            <NavLink to="/articles">
              <Logo className="top_header_logo"></Logo>
            </NavLink>
          </div>

          <div className="navbar_links">
            <div className="container_moon">
              <Moon className="header_moon" onClick={this.onMoonClick}></Moon>
            </div>
            {TokenService.getAuthToken() ? (
              <div className="container_header_logged_in_links">
                <NavLink
                  to="/account"
                  role="navigation"
                  className="navbar_link_account"
                >
                  My Account
                </NavLink>
                <NavLink
                  to="/login"
                  role="navigation"
                  onClick={this.handleLogoutClick}
                  className="navbar_link_login"
                >
                  Logout
                </NavLink>
              </div>
            ) : (
              <div className="container_header_logged_out_links">
                <NavLink
                  to="/login"
                  role="navigation"
                  className="navbar_link_login"
                >
                  Login
                </NavLink>
                <NavLink to="/signup" className="navbar_link_signup">
                  Sign Up
                </NavLink>
              </div>
            )}
          </div>
        </div>
      </div>
    );
  }
}

Header.propTypes = {
  lightsOff: PropTypes.bool,
  toggleLights: PropTypes.func
};

export default Header;
