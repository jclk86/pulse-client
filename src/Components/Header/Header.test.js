import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Header from "./Header";

describe(`Header Component`, () => {
  it("renders Header component without issues", () => {
    const wrapper = shallow(<Header></Header>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
