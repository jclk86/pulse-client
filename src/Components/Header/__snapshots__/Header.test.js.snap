// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders Header component without issues 1`] = `
<div
  className="header"
>
  <div
    className="container_header_content"
  >
    <div
      className="container_logo"
    >
      <NavLink
        to="/articles"
      >
        <Logo
          className="top_header_logo"
        />
      </NavLink>
    </div>
    <div
      className="navbar_links"
    >
      <div
        className="container_header_logged_out_links"
      >
        <NavLink
          className="navbar_link_login"
          role="navigation"
          to="/login"
        >
          Login
        </NavLink>
        <NavLink
          className="navbar_link_signup"
          to="/signup"
        >
          Sign Up
        </NavLink>
      </div>
    </div>
  </div>
</div>
`;
